proxy: Remove oxyproxy, CILIUM_USE_ENVOY.

jrajahalme opened the pull request:
Now that we always test with Envoy we can remove the oxy proxy and
simplify by removing the cilium-agent command line argument
"envoy-proxy" and the use of the environment variable
"CILIUM_USE_ENVOY".

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
Remove oxyproxy and make Envoy the default proxy
```

jrajahalme committed:
proxy: Remove oxyproxy, CILIUM_USE_ENVOY.

Now that we always test with Envoy we can remove the oxy proxy and
simplify by removing the cilium-agent command line argument
"envoy-proxy" and the use of the environment variable
"CILIUM_USE_ENVOY".

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

manalibhutiyani reviewed (approved):
Reviewed only the proxy redirect part. Not the ct tests.

jrajahalme committed:
envoy: Disable unit test by default.

Ginkgo unit tests are run without a preceding build step. Disable
envoy unit test by default to get around this. We should have a
corresponding runtime test instead.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aanm commented on daemon/main.go:
```diff
@@ -336,9 +335,6 @@ func init() {
 	flags.String("enable-policy", endpoint.DefaultEnforcement, "Enable policy enforcement")
 	flags.BoolVar(&enableTracing,
 		"enable-tracing", false, "Enable tracing while determining policy (debugging)")
-	flags.BoolVar(&useEnvoy,
```
What happens to current users that are running cilium with `--envoy-proxy`? I think this flag should still exist either with `(deprecated)` and we remove it on the next version or we market it as an hidden flag in viper.

tgraf replied:
I think removal of the flag is safe pre 1.0. Are you worried about the error if the flag is still passed in?

aanm replied:
correct, there will be an error like:
```
$ cilium-agent --envoy-proxy
Error: unknown flag: --envoy-proxy
```

tgraf replied:
I'm fine to keep it around for another release but we should remove it pre 1.0.

jrajahalme replied:
@aanm Do we have an example of how to hide the flag somewhere?

jrajahalme replied:
I'll try with ```flags.MarkHidden("envoy-proxy")```

aanm replied:
yes, I think the `MarkHidden` does the trick we are using for `cmdref`

jrajahalme committed:
daemon: Keep "--envoy-proxy" one release longer.

Keep the "--envoy-proxy" command line option for the cilium-agent, but
hide it and mark it deprecated. This will prevent existing
configurations from breaking on upgrade.

Requested-by: André Martins <andre@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

raybejjani reviewed (approved):
approved for docs

ianvernon reviewed (approved):
LGTM! Hooray!!
